### PR TITLE
CSS dataset format (import only)

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Summary

- Added Common Semantic Segmentation dataset format
- Updated `mask_tools.unpaint_mask()` error messages in case of unknown color in the mask

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: site/content/en/docs/formats/css.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+To add classes, you should use [`dataset_meta.json`](/docs/user-manual/supported_formats/#dataset-meta-file).
+
+To make sure that the selected dataset has been added to the project, you can
+run `datum project info`, which will display the project information.
+
+## Export to other formats
+
+Datumaro can convert a CSS dataset into any other format [Datumaro supports](/docs/user-manual/supported_formats/).
+To get the expected result, convert the dataset to formats
+that support the segmentation task (e.g. PascalVOC, CamVID, CItyscapes, etc.)
```
Comment (IRDonch) [inline site/content/en/docs/formats/css.md]:
```suggestion
that support the segmentation task (e.g. PASCAL VOC, CamVid, CItyscapes, etc.)
```

File: tests/assets/common_segmentation_dataset/dataset/dataset_meta.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+  "labels": [
+    "Void",
+    "Animal",
+    "Archway",
+    "Bicyclist",
+    "Child",
+    "Road"
+  ],
+  "label_map": {
```
Comment (IRDonch) [inline tests/assets/common_segmentation_dataset/dataset/dataset_meta.json]: Do we really need both `labels` and `label_map`?

File: site/content/en/docs/formats/css.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
+title: 'Common Semantic Segmentation'
+linkTitle: 'Common Semantic Segmentation'
+description: ''
+weight: 2
+---
+
```
Comment (IRDonch) [inline site/content/en/docs/formats/css.md]: This seems to be lacking background information. Where did this format come from?

File: datumaro/plugins/common_segmentation_format.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    else:  # only copy defined colors
+        label_id = lambda label: label_categories.find(label)[0]
+        colormap = {label_id(name): (desc[0], desc[1], desc[2]) for name, desc in label_map.items()}
+    mask_categories = MaskCategories(colormap)
+    mask_categories.inverse_colormap  # pylint: disable=pointless-statement
+    categories[AnnotationType.mask] = mask_categories
+    return categories
+
+
+class CommonSegmentationExtractor(SourceExtractor):
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: I'm a bit concerned that the word "semantic" has disappeared from the name of the format. Why didn't you include it?

File: datumaro/plugins/common_segmentation_format.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        super().__init__(subset=subset)
+
+        self._image_prefix = image_prefix
+        self._mask_prefix = mask_prefix
+
+        if has_meta_file(path):
+            label_map = parse_meta_file(path)
+            self._categories = make_categories(label_map)
+        else:
+            meta_file = glob.glob(osp.join(path, "**", DATASET_META_FILE), recursive=True)
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: This globbing doesn't match the description of the format, according to which the meta file must be in the root. Why is it needed?

File: datumaro/plugins/common_segmentation_format.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+                self._categories = make_categories(label_map)
+
+        self._items = list(self._load_items(path).values())
+
+    def _load_items(self, path):
+        items = {}
+
+        image_dir = osp.join(path, "**", CommonSegmentationPath.IMAGES_DIR)
+
+        if osp.isdir(image_dir):
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: This will almost certainly be false, since `image_dir` has `/**/` in it. Seems like a bug.

File: datumaro/plugins/common_segmentation_format.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            images = {
+                osp.splitext(osp.relpath(p, image_dir))[0].replace("\\", "/")[
+                    len(self._image_prefix) :
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: Shouldn't this also filter out images that _don't_ have the given prefix?

File: datumaro/plugins/common_segmentation_format.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+                ]: p
+                for p in find_images(image_dir, recursive=True)
+            }
+        else:
+            images = {}
+
+        for mask_path in glob.glob(
+            osp.join(path, "**", CommonSegmentationPath.MASKS_DIR, f"{self._mask_prefix}*.*"),
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: 1. Shouldn't masks always be PNG? You even have the `IMAGE_EXT` constant defined for it.
2. `_mask_prefix` should be escaped with `glob.escape`.

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: site/content/en/docs/formats/css.md (hunk 3 above)
Comment (yasakova-anastasia) [reply site/content/en/docs/formats/css.md]: I took the description of this format in AC. We can add a link to [it](https://github.com/openvinotoolkit/open_model_zoo/blob/master/tools/accuracy_checker/openvino/tools/accuracy_checker/annotation_converters/common_semantic_segmentation.py).

File: datumaro/plugins/common_segmentation_format.py (hunk 4 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: I decided that the name `CommonSemanticSegmentation` is too long. 

We can use the CSS name, but since the format is not very well known, it will be difficult for most users to understand what it is for by this name.

File: datumaro/plugins/common_segmentation_format.py (hunk 5 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: This is necessary to support use cases with folder nesting. This is present in WB datasets.

File: datumaro/plugins/common_segmentation_format.py (hunk 6 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: You are right, thanks. I fixed it.

File: datumaro/plugins/common_segmentation_format.py (hunk 7 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: Added a condition for this.

File: datumaro/plugins/common_segmentation_format.py (hunk 8 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: 1. Masks can be not only PNG. This information is in the description of the format in AC.
2. Added `glob.escape`

File: tests/assets/common_segmentation_dataset/dataset/dataset_meta.json (hunk 2 above)
Comment (yasakova-anastasia) [reply tests/assets/common_segmentation_dataset/dataset/dataset_meta.json]: No. Removed `labels`.

File: site/content/en/docs/formats/css.md (hunk 1 above)
Comment (yasakova-anastasia) [reply site/content/en/docs/formats/css.md]: Fixed

Comment (IRDonch) [reply site/content/en/docs/formats/css.md]: You only fixed it half-way.

File: site/content/en/docs/formats/css.md (hunk 3 above)
Comment (IRDonch) [reply site/content/en/docs/formats/css.md]: Was the format created by AC? If so, it would make sense to document that and add a link to the AC documentation.

File: datumaro/plugins/common_segmentation_format.py (hunk 4 above)
Comment (IRDonch) [reply datumaro/plugins/common_segmentation_format.py]: Given that in AC this format is referred to as `common_semantic_segmentation`, I think we should do the same.

File: datumaro/plugins/common_segmentation_format.py [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+        self._items = list(self._load_items(path).values())
+
+    def _load_items(self, path):
+        items = {}
+
+        image_dir = osp.join(path, CommonSegmentationPath.IMAGES_DIR)
+
+        if not osp.isdir(image_dir):
+            image_dir = glob.glob(
+                osp.join(path, "**", CommonSegmentationPath.IMAGES_DIR), recursive=True
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: Do we really need to search for the image and annotation directories like that? I understand the nested folder support, but shouldn't the image and annotation directories at least be in the same directory as the `dataset_meta.json` file? Since we have already located the meta file at this point, we should be able to just locate the images and annotations relative to it.

File: datumaro/plugins/common_segmentation_format.py [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+            if image_dir:
+                image_dir = image_dir[0]
+
+        if osp.isdir(image_dir):
+            images = {
+                osp.splitext(osp.relpath(p, image_dir))[0].replace("\\", "/")[
+                    len(self._image_prefix) :
+                ]: p
+                for p in find_images(image_dir, recursive=True)
+                if osp.basename(p)[: len(self._image_prefix)] == self._image_prefix
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: You can use `.startswith()`.

File: datumaro/plugins/common_segmentation_format.py (hunk 8 above)
Comment (IRDonch) [reply datumaro/plugins/common_segmentation_format.py]: > Masks can be not only PNG. This information is in the description of the format in AC.

OK, but in that case, shouldn't you at least use `find_images` in order to filter out non-image files?

File: datumaro/util/mask_tools.py [hunk 11]
```diff
@@ -64,7 +64,7 @@ def unpaint_mask(painted_mask, inverse_colormap=None):
     if callable(inverse_colormap):
         map_fn = lambda a: inverse_colormap((a >> 16) & 255, (a >> 8) & 255, a & 255)
     else:
-        map_fn = lambda a: inverse_colormap[((a >> 16) & 255, (a >> 8) & 255, a & 255)]
+        map_fn = lambda a: inverse_colormap.get(((a >> 16) & 255, (a >> 8) & 255, a & 255), 0)
```
Comment (IRDonch) [inline datumaro/util/mask_tools.py]: Can you explain this change?

File: tests/test_common_segmentation_format.py [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+from datumaro.plugins.common_segmentation_format import CommonSegmentationImporter, make_categories
+from datumaro.util.test_utils import compare_datasets
+
+from .requirements import Requirements, mark_requirement
+
+DUMMY_DATASET_DIR = osp.join(
+    osp.dirname(__file__), "assets", "common_segmentation_dataset", "dataset"
+)
+
+DUMMY_DATASET_WITH_NON_STANDARD_STRUCTURE = osp.join(
```
Comment (IRDonch) [inline tests/test_common_segmentation_format.py]:
```suggestion
DUMMY_DATASET_WITH_NON_STANDARD_STRUCTURE_DIR = osp.join(
```

File: datumaro/plugins/common_segmentation_format.py [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+    def _load_items(self, path):
+        items = {}
+
+        image_dir = osp.join(path, CommonSegmentationPath.IMAGES_DIR)
+
+        if not osp.isdir(image_dir):
+            image_dir = glob.glob(
+                osp.join(path, "**", CommonSegmentationPath.IMAGES_DIR), recursive=True
+            )
+            if image_dir:
```
Comment (IRDonch) [inline datumaro/plugins/common_segmentation_format.py]: If this is false, then `image_dir` will remain a list, and the next call to `osp.isdir` will crash.

File: site/content/en/docs/formats/css.md (hunk 3 above)
Comment (yasakova-anastasia) [reply site/content/en/docs/formats/css.md]: Added a link to AC documentation for this format.

File: datumaro/plugins/common_segmentation_format.py (hunk 4 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: Renamed to `common_semantic_segmentation`

File: datumaro/plugins/common_segmentation_format.py (hunk 8 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: Fixed

File: datumaro/plugins/common_segmentation_format.py (hunk 9 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: I agree. Fixed it.

File: datumaro/util/mask_tools.py (hunk 11 above)
Comment (yasakova-anastasia) [reply datumaro/util/mask_tools.py]: While testing this format on some dataset, I found this problem. The masks have pixels that could not be identified using the given classes. Such pixels should be defined as background.

File: datumaro/plugins/common_segmentation_format.py (hunk 10 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: Fixed

File: datumaro/plugins/common_segmentation_format.py (hunk 13 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_segmentation_format.py]: Removed this code

File: site/content/en/docs/formats/common_semantic_segmentation.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+To add classes, you should use [`dataset_meta.json`](/docs/user-manual/supported_formats/#dataset-meta-file).
+
+To make sure that the selected dataset has been added to the project, you can
+run `datum project info`, which will display the project information.
+
+## Export to other formats
+
+Datumaro can convert a CSS dataset into any other format [Datumaro supports](/docs/user-manual/supported_formats/).
+To get the expected result, convert the dataset to formats
+that support the segmentation task (e.g. PASCAL VOC, CamVid, CItyscapes, etc.)
```
Comment (zhiltsov-max) [inline site/content/en/docs/formats/common_semantic_segmentation.md]:
```suggestion
that support the segmentation task (e.g. PASCAL VOC, CamVid, Cityscapes, etc.)
```

File: tests/test_common_semantic_segmentation_format.py [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+)
+from datumaro.util.test_utils import compare_datasets
+
+from .requirements import Requirements, mark_requirement
+
+DUMMY_DATASET_DIR = osp.join(
+    osp.dirname(__file__), "assets", "common_semantic_segmentation_dataset", "dataset"
+)
+
+DUMMY_DATASET_WITH_NON_STANDARD_STRUCTURE_DIR = osp.join(
```
Comment (zhiltsov-max) [inline tests/test_common_semantic_segmentation_format.py]: This dataset has custom filenames, but the structure is the same as in the "normal" case. I suggest to rename this dataset to make the difference more clear.

File: datumaro/util/mask_tools.py (hunk 11 above)
Comment (zhiltsov-max) [reply datumaro/util/mask_tools.py]: > Such pixels should be defined as background.

Where this comes from? Why not throw an error?

Also, we don't always have background at id 0 already, so maybe it should be a parameter.

Comment (yasakova-anastasia) [reply datumaro/util/mask_tools.py]: I think that if a pixel does not belong to any presented class, it belongs to the background (it seemed to me that this class was partially needed for this). But I agree that this can be interpreted as an error in the dataset.

> Also, we don't always have background at id 0 already, so maybe it should be a parameter.

Do you mean add a parameter to the `load_mask` function for the background class number?

Comment (IRDonch) [reply datumaro/util/mask_tools.py]: +1 to treating this as an error.

File: tests/test_common_semantic_segmentation_format.py (hunk 15 above)
Comment (yasakova-anastasia) [reply tests/test_common_semantic_segmentation_format.py]: Renamed

File: datumaro/util/mask_tools.py (hunk 11 above)
Comment (yasakova-anastasia) [reply datumaro/util/mask_tools.py]: Added an error for this case.

Comment (zhiltsov-max) [reply datumaro/util/mask_tools.py]: > I think that if a pixel does not belong to any presented class, it belongs to the background 

Why?

> it seemed to me that this class was partially needed for this. But I agree that this can be interpreted as an error in the dataset.

I agree, that it can be true in some cases, like id 255 / color 255,255,255 in VOC masks is used for "boundary" pixels, which are not supposed to be used for accuracy evaluation. But this fact cannot be treated as a common practice, or always supposed.

> Do you mean add a parameter to the load_mask function for the background class number?

No, I'm speaking about the `unpaint_mask` function, which is being discussed in this thread. We could add a parameter to map unknown pixels to background, if it is needed, but it should not be the default behavior.

File: datumaro/plugins/common_semantic_segmentation_format.py [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument("--image-prefix", default="", help="Image prefix (default: '')")
+        parser.add_argument("--mask-prefix", default="", help="Mask prefix (default: '')")
+        return parser
+
+    @classmethod
+    def detect(cls, context: FormatDetectionContext) -> None:
+        context.require_file(f"**/{CommonSemanticSegmentationPath.IMAGES_DIR}/**/*.*")
+        context.require_file(f"**/{CommonSemanticSegmentationPath.MASKS_DIR}/**/*.*")
+        context.require_file(f"**/{DATASET_META_FILE}")
```
Comment (IRDonch) [inline datumaro/plugins/common_semantic_segmentation_format.py]: Similarly to the loading code, this doesn't need to search for images and masks everywhere in the tree. Once you have located the meta file, you can take its parent directory and search there.

File: datumaro/util/mask_tools.py [hunk 17]
```diff
@@ -64,14 +65,22 @@ def unpaint_mask(painted_mask, inverse_colormap=None):
...
 
     painted_mask = painted_mask.astype(int)
     painted_mask = (
         painted_mask[:, :, 0] + (painted_mask[:, :, 1] << 8) + (painted_mask[:, :, 2] << 16)
     )
     uvals, unpainted_mask = np.unique(painted_mask, return_inverse=True)
-    palette = np.array([map_fn(v) for v in uvals], dtype=np.min_scalar_type(len(uvals)))
+    palette = []
+    for v in uvals:
+        color = map_fn(v)
```
Comment (IRDonch) [inline datumaro/util/mask_tools.py]: `map_fn` doesn't return a color, it returns a class index.

File: datumaro/util/mask_tools.py [hunk 18]
```diff
@@ -64,14 +65,22 @@ def unpaint_mask(painted_mask, inverse_colormap=None):
...
+        if color is None:
+            raise DatasetError(
```
Comment (IRDonch) [inline datumaro/util/mask_tools.py]: I don't think `DatasetError` makes sense here, since there is no requirement that the mask must come from a dataset.

Comment (yasakova-anastasia) [reply datumaro/util/mask_tools.py]: Replaced with `AnnotationError`

File: datumaro/plugins/common_semantic_segmentation_format.py (hunk 16 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_semantic_segmentation_format.py]: Fixed

File: datumaro/util/mask_tools.py (hunk 17 above)
Comment (yasakova-anastasia) [reply datumaro/util/mask_tools.py]: Fixed

File: datumaro/util/mask_tools.py (hunk 18 above)
Comment (yasakova-anastasia) [reply datumaro/util/mask_tools.py]: Replaced with `UndeclaredColorError`

Comment (IRDonch) [reply datumaro/util/mask_tools.py]: @zhiltsov-max Please check that the new exception fits into your framework.

**Review by IRDonch — DISMISSED** [review]

File: CHANGELOG.md [hunk 19]
```diff
@@ -15,6 +15,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   (<https://github.com/openvinotoolkit/datumaro/pull/539>)
 - BraTS format (import-only) (.npy and .nii.gz), new `MultiframeImage`
   media type (<https://github.com/openvinotoolkit/datumaro/pull/628>)
+- Common Semantic Segmentation dataset format (import-only)
```
Comment (zhiltsov-max) [inline CHANGELOG.md:18]: Don't you think it's worth making not import-only? I feel, this format could be used extensively.

File: datumaro/plugins/common_semantic_segmentation_format.py [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+
+        self._image_prefix = image_prefix
+        self._mask_prefix = mask_prefix
+
+        self._root_dir = path
+
+        if has_meta_file(path):
+            label_map = parse_meta_file(path)
+            self._categories = make_categories(label_map)
+        else:
```
Comment (zhiltsov-max) [inline datumaro/plugins/common_semantic_segmentation_format.py]: I can't see real difference between these branches. Can we use just the `else` branch? Also, where the meta file can be found, if not in the root?

File: datumaro/plugins/common_semantic_segmentation_format.py [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+            mask = lazy_mask(mask_path, self._categories[AnnotationType.mask].inverse_colormap)
+            mask = mask()  # loading mask through cache
+
+            classes = np.unique(mask)
+            for label_id in classes:
+                annotations.append(
+                    Mask(image=self._lazy_extract_mask(mask, label_id), label=label_id)
+                )
+
+            items[item_id] = DatasetItem(id=item_id, media=image, annotations=annotations)
```
Comment (zhiltsov-max) [inline datumaro/plugins/common_semantic_segmentation_format.py]:
```suggestion
            items[item_id] = DatasetItem(id=item_id, subset=self._subset, media=image, annotations=annotations)
```

File: CHANGELOG.md (hunk 19 above)
Comment (yasakova-anastasia) [reply CHANGELOG.md:18]: This format is now needed only for import for WB.

File: datumaro/util/mask_tools.py [hunk 22]
```diff
@@ -64,14 +65,20 @@ def unpaint_mask(painted_mask, inverse_colormap=None):
     if callable(inverse_colormap):
         map_fn = lambda a: inverse_colormap((a >> 16) & 255, (a >> 8) & 255, a & 255)
     else:
-        map_fn = lambda a: inverse_colormap[((a >> 16) & 255, (a >> 8) & 255, a & 255)]
+        map_fn = lambda a: inverse_colormap.get(((a >> 16) & 255, (a >> 8) & 255, a & 255), None)
```
Comment (zhiltsov-max) [inline datumaro/util/mask_tools.py:70]: Could you add docs and tests for this function?

File: site/content/en/docs/formats/common_semantic_segmentation.md [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+## Import Common Semantic Segmentation dataset
+
+A Datumaro project with a CSS source can be created in the following way:
+
+``` bash
+datum create
+datum import --format common_semantic_segmentation <path/to/dataset>
+```
+
+Extra options for adding a source in the COCO format:
```
Comment (zhiltsov-max) [inline site/content/en/docs/formats/common_semantic_segmentation.md]:
```suggestion
Extra import options:
```

COCO (panoptic) doesn't seem to be related to this format, because it stores masks in a different encoding. Also, they won't be located by this format, if the dataset follows the default Datumaro COCO layout.

File: site/content/en/docs/formats/common_semantic_segmentation.md [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+    │   ├── <img1>.png
+    │   ├── <img2>.png
+    │   └── ...
+    └── masks/
+        ├── <img1>.png
+        ├── <img2>.png
+        └── ...
+```
+
+To add classes, you should use [`dataset_meta.json`](/docs/user-manual/supported_formats/#dataset-meta-file).
```
Comment (zhiltsov-max) [inline site/content/en/docs/formats/common_semantic_segmentation.md]:
```suggestion
To describe classes and colors, you should use [`dataset_meta.json`](/docs/user-manual/supported_formats/#dataset-meta-file).
```

File: datumaro/util/mask_tools.py (hunk 18 above)
Comment (zhiltsov-max) [reply datumaro/util/mask_tools.py]: I like the error itself, but I'm not sure an `InvalidAnnotationError` (which is inherited from `DatasetImportError`) should be used in this context. I'd rather just throw a regular `KeyError` and handled in on the calling side.

File: CHANGELOG.md (hunk 19 above)
Comment (zhiltsov-max) [reply CHANGELOG.md:18]: How WB is going to obtain such datasets? This format is crafted for OpenVINO and it is essential, that all our projects can work with it. Someone needs to be able to write in this format.

File: datumaro/plugins/common_semantic_segmentation_format.py (hunk 20 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/common_semantic_segmentation_format.py]: The meta file can be in a subfolder in some cases (located together with the images and masks folders)

File: CHANGELOG.md (hunk 19 above)
Comment (yasakova-anastasia) [reply CHANGELOG.md:18]: I agree with this. We will need to support export in this format. But at the moment, the format is needed only for detection.
I think that the export task can wait.

File: datumaro/util/mask_tools.py (hunk 22 above)
Comment (yasakova-anastasia) [reply datumaro/util/mask_tools.py:70]: Added test for this function

**Review by zhiltsov-max — APPROVED** [review]
